File: 14-LongestCommonPrefix/14-LongestCommonPrefix.java
// Last updated: 25/03/2026, 15:17:52
class Solution {
    public String longestCommonPrefix(String[] strs) {

        int minLen = strs[0].length();  

        for (int i = 0; i < strs.length - 1; i++) {
            String first = strs[i];
            String second = strs[i + 1];

            int j = 0;
            int min = Math.min(first.length(), second.length());

            while (j < min) {

                
                if (first.charAt(j) != second.charAt(j)) {
                    break;
                }

               
                if (j == minLen) {
                    break;
                }

                j++;
            }

            
            minLen = Math.min(minLen, j);
        }

        return strs[0].substring(0, minLen);
    }
}

File: 14-LongestCommonPrefix/14-LongestCommonPrefix.cpp
// Last updated: 14/05/2026, 16:01:29
1class Solution {
2public:
3    string longestCommonPrefix(vector<string>& strs) {
4
5        string prefix=strs[0];
6
7        for(int i=1;i<strs.size();i++){
8            
9            while(strs[i].find(prefix)!=0){
10                prefix.pop_back();
11            }
12        }
13
14        return prefix;
15        
16    }
17};